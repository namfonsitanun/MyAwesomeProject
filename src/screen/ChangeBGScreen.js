import React, { Component } from 'react'
import { View, StyleSheet, TextInput } from 'react-native'

export default class ChangeBGScreen extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			bgcolor: '#FFFFFF',
		}
	}

	render() {
		return (
			<View style={styles(this.state.bgcolor).container}>
				<TextInput
					style={styles.input}
					placeholder="color hex code"
					onChangeText={bgcolor => this.setState({ bgcolor })}
				/>
			</View>
		)
	}
}

const styles = color =>
	StyleSheet.create({
		container: {
			// color: "black",
			flex: 1,
			justifyContent: 'center',
			alignItems: 'center',
			backgroundColor: color,
		},
		input: {
			textAlign: 'center',
		},
	})
