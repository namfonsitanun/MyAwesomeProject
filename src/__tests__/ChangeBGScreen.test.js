import React from 'react'
import 'react-native'
import ChangeBGScreen from '../screen/ChangeBGScreen.js'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'styled-components-test-utils/lib/jest'

describe('<ChangeBGScreen>', () => {
  it('Should have black BG', () => {
    const props = {
      bgcolor: 'black'
    }
    const bg = renderer.create(<ChangeBGScreen {...props} />).toJSON()
    // expect(bg).toMatchSnapshot()
    expect(bg).toHaveStyleRule('backgroundColor', 'black')
  })
})
