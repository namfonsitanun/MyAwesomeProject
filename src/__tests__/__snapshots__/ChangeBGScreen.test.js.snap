// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangeBGScreen> Should have black BG 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="color hex code"
  />
</View>
`;
